nodes/src/nodes/llm_ibm_watson/test_location_injection.py

```
@@ -0,0 +1,122 @@
+
+    @pytest.mark.parametrize('payload', [
+        'attacker.com/x#',               # fragment injection
+        'attacker.com/x?',               # query injection
+        'evil.com:443/path#',            # port + fragment
+        'evil.com@legitimate.com',       # userinfo injection
+        '../../../etc/passwd',           # path traversal
+        'us-south.ml.cloud.ibm.com#',   # full domain with fragment
+        'attacker.com\\@ibm.com',        # backslash injection
+    ])
+    def test_ssrf_payload_rejected(self, payload):
+        with pytest.raises(ValueError):
```

Ruff (0.15.6)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)